README.md

```diff
@@ -1,17 +1,261 @@
+they are getting, and what to do with it.
+
+The code for a redirect is `302`. When a browser makes a request, and
+gets a `302` code, it knows that it need to follow the "redirect" to the
+given link and load that page next.
+
+AJAX requests don't follow redirects because they *can't*. If they did,
+then you would lose any code that happens in the `done()` function and
+just go to a new page that doesn't have that JavaScript code.
+
+So when an AJAX request sees a 302, it chooses not to follow the link
+and instead just takes whatever the rendered data is.
+
+Rails handles HTTP codes for us when we do HTML requests, but when we
+build JSON APIs, we need to be cognizant of using the appropriate codes
+to let the client know what to expect.
+
+We'll see this in action shortly.
+
+### Rendering the Response
+
+What we really want to do is get a JSON representation of the post we
+just created so that we can use it to display the new post without
+redirecting or refreshing the page.
+
+Let's get into the controller, get rid of that redirect, and use the
+ActiveModel::Serializer that we already have.
+
+```ruby
+# posts_controller.rb
+# ...
+  def create
+    @post = Post.create(post_params)
+    render json: @post, status: 201
+  end
+```
+
+**Note::** We are specifying the `status: 201` for this request, rather
+than the standard `200`, which means `OK`. Technically, this is a
+successful request and could be considered `200`, but we want to
+specify what happened more granularly, and `201` means that the resource
+was `created`.
+
+With our controller code in place, we need to make some additions to our
+template to handle the response and display it on the page.
+
+```erb
+<%= form_for(@post) do |f| %>
+  <div class="field">
+    <%= f.label :title %><br>
+    <%= f.text_field :title %>
+  </div>
+  <div class="field">
+    <%= f.label :description %><br>
+    <%= f.text_field :description %>
+  </div>
+  <div class="actions">
+    <%= f.submit %>
+  </div>
+<% end %>
+
+<div id="postResult">
+  <h2 id="postTitle"></h2>
+  <p id="postBody"></p>
+</div>
+
+<script type="text/javascript" charset="utf-8">
+  $(function () {
+    $('form').submit(function(event) {
+      //prevent form from submitting the default way
+      event.preventDefault();
+
+      var values = $(this).serialize();
+
+      var posting = $.post('/posts', values);
+
+      posting.done(function(data) {
+        var post = data["post"];
+        $("#postTitle").text(post["title"]);
+        $("#postBody").text(post["description"]);
+      });
+    });
+  });
+</script>
+```
+
+Very similar to what we've been doing when we use AJAX to `get()` a
+resource, we just parse the JSON and add it to the DOM. To keep it
+simple, we just have one `<DIV>` that has placeholders for the data, and
+we fill it in when we get that response.
+
+## Summary
+
+We've learned how to POST a form using jQuery, render a response, and
```

Comment:
Cool! I think this lesson does a great job of being entertaining and also getting students to use AJAX effectively.